### ISSUE TITLE
Merge segments before calculation

### ISSUE THREAD

**Issue by tordans** [osmberlin/osm-parking-processing#66]
<img width="222" alt="image" src="https://user-images.githubusercontent.com/111561/225119626-5ca3e346-861a-447a-b703-c26c659c2042.png">

**[Location](https://radverkehrsatlas.de/regionen/parkraum?lat=52.48172&lng=13.42966&zoom=18&theme=parking&bg=default&config=!(i~parking~topics~!(i~parking~a~~s~!(i~default~a~_F)(i~presence~a)(i~raw~a~_F))(i~parkingPoints~a~_F~s~!(i~default~a))(i~parkingAreas~a~~s~!(i~default~a)(i~position-separate~a~_F))(i~parkingDebug~a~~s~!(i~default~a))(i~parkingStats~a~_F~s~!(i~default~a~~f~!(i~admin*_level~o~!(i~*4~a~_F)(i~*9~a)(i~*10~a~_F)))(i~length~a~_F~f~!(i~admin*_level~o~!(i~*4~a~_F)(i~*9~a)(i~*10~a~_F))))(i~landuse~a~_F~s~!(i~default~a)))~)**

This looks like a nice edge case:
The OSM way that the "Missing data" Segment belongs to is https://www.openstreetmap.org/way/1133008268. This one gets split by the driveway.

The left over piece is too small for processing. However, we should in fact merge it first with the line next to it which has the same parking tags. Together they would likely have space for 1 more car…

**Comment by SupaplexOSM** [comment]
Ja, das sollten wir auf jeden Fall tun. Das ist eigentlich kein Edge Case, weil diese Art von Fehler jederzeit passiert, wenn sich Attribute der Straße oder einer Parkstreifenseite ändern, was bei detailliert erfassten Daten oft der Fall ist (etwas weiter südlich z.B. auf der Flughafenstraße gibt es auch viele kurze Segmente nebeneinander, obwohl einige davon zusammengehören, weil mal auf der einen, mal auf der anderen Seite die Parkbeschränkungen wechseln).

Das Pythonscript "dissolved" einfach alle Parksegmente mit gleichen Attributen, wodurch nebeneinanderliegende "gleiche" Segmente verschmelzen (siehe Zeile [1879ff.](https://github.com/SupaplexOSM/street_parking.py/blob/main/street_parking.py#L1879)) – funktioniert gut.

Es gibt im Pythonscript allerdings noch einen Edge Case, der momentan nicht abgefangen wird: Die verschmelzenden/benachbarten Segmente sollten in die gleiche Richtung zeigen, sonst könnten theoretisch ein "linker" und ein "rechter" Parkstreifen gemerged werden, was zu einem fehlerhaften "Tausch" der Eigenschaften an einem der Parkstreifen führen würde. Irgendwo im Schillerkiez gibt es eine Stelle, wo das passiert (zwei an der gleichen Kreuzung startende, aber in verschiedene Richtungen zeigende Linien, die jeweils auf der linken und rechten Seite die gleichen Parkattribute haben – was sich durch die verdrehte Richtung in der Realität aber auf verschiedene Straßenseiten bezieht).

**Comment by joshinils** [comment]
Test ob hier eine preview erscheint;
https://github.com/SupaplexOSM/street_parking.py/blob/c6757375bbbfe2ed4037ebddbd7a500e8a580f45/street_parking.py#L1879
